### PR TITLE
[DT-779] Clear selected filters code

### PR DESCRIPTION
**Pull request by s-rubenstein**
### Addresses
Adds a clear filters button

### Summary

https://github.com/user-attachments/assets/52904ca1-8b6e-43ba-b4fd-5a4cc84104ac

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Comment by rushtong** [comment]
I think this interferes with the search bar. If I enter a value there and clear the search, the dataset list remains unchanged.

**Comment by s-rubenstein** [comment]
Good call! Clear search definitely seems affected. I'll poke around

File: src/components/data_search/DatasetFilterList.jsx [hunk 1]
```diff
@@ -6,7 +6,7 @@ import ListItemButton from '@mui/material/ListItemButton';
 import ListItemIcon from '@mui/material/ListItemIcon';
 import ListItemText from '@mui/material/ListItemText';
 import Divider from '@mui/material/Divider';
-import { Typography } from '@mui/material';
+import { Button, Typography } from '@mui/material'
```
Comment (rushtong) [inline src/components/data_search/DatasetFilterList.jsx]: nit: missing semicolon

**Review by rushtong — APPROVED** [review]
Looks reasonable 👍🏽

**Review by fboulnois — CHANGES_REQUESTED** [review]
The search no longer works as expected:

1. Enter a keyword like "All"
2. Clicking "Clear Search" has no effect
3. Manually removing the search text has no effect

**Comment by s-rubenstein** [comment]
@fboulnois that is not what I see on my local - maybe we can verify we are running the same version?

**Comment by s-rubenstein** [comment]
https://github.com/user-attachments/assets/316aaecf-8fbe-4eec-a649-27a381009b78

**Review by fboulnois — APPROVED** [review]
👍

**Pull request merged**